dask_kubernetes/core.py

```diff
@@ -500,27 +514,58 @@ def f(to_close):
                     raise
 
     async def _cleanup_terminated_pods(self, pods):
-        terminated_phases = {"Succeeded", "Failed"}
+        terminated_phases = {PodStatus.succeeded.value, PodStatus.failed.value}
         terminated_pods = [p for p in pods if p.status.phase in terminated_phases]
         await self._delete_pods(terminated_pods)
-        return [p for p in pods if p.status.phase not in terminated_phases]
+        pods = [p for p in pods if p.status.phase not in terminated_phases]
+        return pods
 
     def scale_up(self, n, **kwargs):
-        self.scheduler.loop.add_callback(self._scale_up, n, **kwargs)
+        self._manual_scale_target = n
+        self.scheduler.loop.add_callback(self._scale_up, **kwargs)
 
-    async def _scale_up(self, n, pods=None, **kwargs):
+    async def check_status(self, status_check=PodStatus.running):
+        while True:
+            pods = await self._pods()
+            phases = {p.metadata.name: p.status.phase for p in pods}
+            result = [status_check.value == status for name, status in phases.items()]
+
+            # throttle loop
+            await asyncio.sleep(0.5)
+            if result is None:
+                continue
+            elif sum(result) < self._manual_scale_target:
+                continue
+            else:
+                return True
+
+    async def status_with_timeout(self, timeout):
+        t = asyncio.create_task(self.check_status())
+        done, pending = await asyncio.wait({t}, timeout=timeout)
```

Comment:
I'm curious, is there a nicer way to do this?   Does asyncio require the `create_task` function to be run before `wait`?  (I don't know as much about asyncio as I do about tornado)

Reply:
switched to a queueing system